File: C11/백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.c
#include <stdio.h>

int main(void){
	
	int N, M;
	scanf("%d %d", &N, &M);
	
	int arr[N + 1];
	for(int i = 1; i <= N; i++){
		scanf("%d", &arr[i]);
	}
	
	int prefixSum[N + 1];
	for(int i = 1; i <= N; i++){
		prefixSum[i] = prefixSum[i - 1] + arr[i];
	}
	
	for(int i = 0; i < M; i++){
		int start, end;
		scanf("%d %d", &start, &end);
		printf("%d\n", prefixSum[end] - prefixSum[start - 1]);
	}
	
	return 0;
}

File: Java/백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        int N = Integer.parseInt(input[0]); //수의 개수
        int M = Integer.parseInt(input[1]); //구해야 하는 횟수

        int[] arr = new int[N + 1];
        String[] inputArr = br.readLine().split(" ");
        for(int i = 1; i <= N; i++){
            arr[i] = Integer.parseInt(inputArr[i - 1]);
        }

        int[] prefixSum = new int[N + 1];
        for (int i = 1; i <= N; i++){
            prefixSum[i] = prefixSum[i - 1] + arr[i];
        }

        for (int i = 0; i < M; i++) {
            String[] range = br.readLine().split(" ");
            int start = Integer.parseInt(range[0]);
            int end = Integer.parseInt(range[1]);

            System.out.println(prefixSum[end] - prefixSum[start - 1]);
        }
    }
}